ui/jstests/test_manage_taxonomies.jsx

```diff
@@ -91,4 +345,348 @@ define(['QUnit', 'jquery', 'setup_manage_taxonomies', 'reactaddons',
+        assert.equal(
+          component.state.vocabularies.length,
+          0
+        );
+        assert.equal(
+          component.state.learningResourceTypes.length,
+          0
+        );
+        waitForAjax(2, function() {
+          assert.equal(
+            component.state.vocabularies.length,
+            1
+          );
+          assert.equal(
+            component.state.vocabularies[0].terms.length,
+            2
+          );
+          assertAddTermCommon(assert, component);
+          // Adding second Vocabulary to make sure term added to correct vocabulary
+          component.addVocabulary(vocabularyWithoutTerms);
+          assert.equal(
+            component.state.vocabularies.length,
+            2
+          );
+          var inputGroup = React.addons.TestUtils.
+          scryRenderedDOMComponentsWithClass(
+            component,
+            'input-group'
+          );
+          assert.equal(inputGroup.length, 2);
+          var textbox = React.addons.TestUtils.
+          findRenderedDOMComponentWithTag(
+            inputGroup[0],
+            'input'
+          );
+          assert.equal(
+            component.state.vocabularies[0].terms.length,
+            2
+          );
+          React.addons.TestUtils.Simulate.keyUp(textbox, {key: "Enter"});
+          waitForAjax(1, function() {
+            assert.equal(
+              component.state.vocabularies[0].terms.length,
+              3
+            );
+            done();
+          });
+        });
+      };
+      React.addons.TestUtils.renderIntoDocument
+      (
+        <TaxonomyComponent
+          vocabularies={vocabularies}
+          repoSlug="repo"
+          ref={afterMount}
+        />
+      );
+    }
+  );
+  QUnit.test('Assert that add vocabulary works in TaxonomyComponent',
+    function(assert) {
+      assert.ok(TaxonomyComponent, "class object not found");
+      var vocabularies = [
+        {
+          "vocabulary": vocabulary,
+          "terms": vocabulary.terms
+        }
+      ];
+      var vocabularyWithoutTerms = {
+        "id": 2,
+        "slug": "difficulty2",
+        "name": "difficulty2",
+        "description": "easy",
+        "vocabulary_type": "f",
+        "required": false,
+        "weight": 2147483647,
+      };
+      var done = assert.async();
+      var afterMount = function(component) {
+        waitForAjax(2, function() {
+          assert.equal(
+              component.state.vocabularies.length,
+              1
+          );
+          var formNode = React.addons.TestUtils.
+            findRenderedDOMComponentWithClass(
+              component,
+              'form-horizontal'
+            );
+          assert.ok(formNode);
+          //test form submission
+          var inputNodes = React.addons.TestUtils.
+            scryRenderedDOMComponentsWithTag(
+              formNode,
+              'input'
+            );
+          inputNodes[0].value = "TestA";
```

Comment:
Note that the result of `findRenderedDOMComponentWithClass` is actually a React component. If you wanted to deal with the DOM you need to call `inputNodes[0].getDOMNode()`